src/runtime_cache_table.js

```diff
@@ -0,0 +1,34 @@
+#if CACHE_WASM_TABLE
+// Wrap the wasm table with a cache, to avoid repeatedly calling table.get().
+// That operation is fairly heavyweight in VMs.
+// N.B.: This cache assumes that the wasm does not modify the table, which is
+//       not necessarily true in relocatable builds (a side module segment may
+//       add items), or if we some day use wasm table operations in wasm.
+var originalWasmTable = wasmTable;
+wasmTable = {
+  _cache: [],
+  'length': wasmTable.length,
+  'get': function(i) {
+    if (i >= wasmTable._cache.length) {
```

Comment:
doing `var c = this._cache;` or `var c = wasmTable._cache` at the top (if not caring about `this` correctness), you can save code size on multiple references on `wasmTable._cache` below. (Closure won't do this kind of outlining)

Reply:
oh, there is also a bug that

```js
if (i >= wasmTable._cache.length) {
  wasmTable._cache.length = i;
}
```
produced an array one index too small, so it should assign `i+1` to `length`?